sormas-api/src/main/java/de/symeda/sormas/api/contact/ContactReferenceDto.java

```diff
@@ -53,12 +53,20 @@ public ContactReferenceDto(String uuid, String contactFirstName, String contactL
 
 		setUuid(uuid);
 		this.contactName = new PersonName(contactFirstName, contactLastName);
-		this.caseName = new PersonName(caseFirstName, caseLastName);
+
+		if (caseFirstName != null && caseLastName != null) {
+			this.caseName = new PersonName(caseFirstName, caseLastName);
+		}
```

Comment:
> I have not, however, understood why the pseudonymization has to be skipped.

It's not a must to skip the pseudonymization but it doesn't make sense to call it on a null value.
The idea is that if the case name is null that means the contact is not assigned to a case, so there is no need to pseudonomize the case name.

> It might be simpler to have a static method `PersonName::of` that takes care of the null check and always use that instead of the constructor

Doesn't make too much sense because contacts always have firstName and lastName, so only the case first and last name should be checked and the `PersonName` class is an inner class of `ContactReferenceDto` and used only in that class